base/array.jl

```diff
@@ -109,6 +137,19 @@ function isassigned(a::Array, i::Int...)
     ccall(:jl_array_isassigned, Cint, (Any, UInt), a, ii) == 1
 end
 
+"""
+    Base.selectorbytes(A::Array{T, N}) -> Array{UInt8, N}
+
+For an Array with `isbits` Union elements, return the "selector bytes" that indicate the index
+of the type for each array element, i.e. the type of `A[1]` is `Base.uniontypes(eltype(A))[Base.selectorbytes(A)[1] + 1]`.
+**NOTE**: The actual array selector bytes are returned, meaning if individual elements are modified, the original array will reflect those changes.
+Setting selector bytes to invalid or out-of-bounds type indexes may corrupt the original array.
+"""
+function selectorbytes(a::Array{T, N}) where {T, N}
+    isbitsunion(T) || return UInt8[]
+    return unsafe_wrap(Array{UInt8, N}, convert(Ptr{UInt8}, pointer(a)) + length(a) * elsize(a), size(a))
```

Comment:
Actually, since it doesn't appear we use this anywhere, I would prefer not to provide it. It violates our aliasing and gc assumptions. It must not be getting tested fully either right now, since the bytes aren't at `length(a) * elsize(a)` (we don't expose any way to get at the actual location currently – we would need to export something new from C to return it)